Sources/Testing/Running/Configuration.swift

```diff
@@ -162,4 +162,33 @@ public struct Configuration: Sendable {
       testFilter = selection.contains
     }
   }
+
+  /// A function that handles filtering test cases.
+  ///
+  /// - Parameters:
+  ///   - testCase: The test case to be filtered.
+  ///   - test: The test which `testCase` is associated with.
+  ///
+  /// - Returns: A Boolean value representing if the test case satisfied the
+  ///   filter.
+  @_spi(ExperimentalParameterizedTesting)
+  public typealias TestCaseFilter = @Sendable (_ testCase: Test.Case, _ test: Test) -> Bool
```

Comment:
Should this take a single argument of type `Event.Context` (or whatever we called it)?

Reply:
I don't think they're the same thing semantically: We only form an `Event.Context` once we have an `Event`, but here we're deciding whether to include a test case, which happens before we begin creating events for it. Plus this aligns more with `TestFilter` typealias which accepts a `Test`.